src/app/shareUrl/sharelink-url.ts

```diff
@@ -0,0 +1,44 @@
+// Copyright 2022 The Triple Banana Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+export class ShareLink {
+  static readonly SHARE_LINK_PARAM = "?sharelink="
+
+  private protocol: string = ""
+  private shareUrl: string = ""
+  private urlWithoutProtocol: string = ""
+
+  constructor(bananaShareUrl: string) {
+    const shareLink = this.parseShareLink(bananaShareUrl)
+    if (!shareLink) return
+
+    this.shareUrl = shareLink
+    this.protocol = this.parseProtocol(shareLink)
```

Comment:
Shouldn't we allow only for protocol `https` and `http`?
I mean how we can handle when another protocol is coming.